src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al

```diff
@@ -163,12 +164,21 @@ codeunit 304 "No. Series - Impl."
             UsageDate := WorkDate();
 
         // Find the No. Series Line closest to the usage date
-        NoSeriesLine.Reset();
-        NoSeriesLine.SetCurrentKey("Series Code", "Starting Date");
-        NoSeriesLine.SetRange("Series Code", NoSeriesCode);
-        NoSeriesLine.SetRange("Starting Date", 0D, UsageDate);
-        NoSeriesLine.SetRange(Open, true);
-        if (NoSeriesLine."Line No." <> 0) and (NoSeriesLine."Series Code" = NoSeriesCode) then begin
+        NoSeriesLine2.Reset();
```

Comment:
NoSeriesLine may be temporary and have lines that have been closed (but are not closed in the database which this looks up). If it's temporary, we should be looking up in NoSeriesLine

Reply:
As you can see a bit below, NoSeriesLine2 is only used to check whether the line we would find matches in start date to the line given. All other code still acts on NoSeriesLine ..